File: engine/engine-model/src/main/scala/client/MapleMount.scala
package client

class MapleMount(var itemId: Int, var skillId: Int) {
  private var _tiredness: Int = 0

  var level: Int = 1

  var exp: Int = 0

  var active: Boolean = true

  /**
   * 1902000 - Hog
   * 1902001 - Silver Mane
   * 1902002 - Red Draco
   * 1902005 - Mimiana
   * 1902006 - Mimio
   * 1902007 - Shinjou
   * 1902008 - Frog
   * 1902009 - Ostrich
   * 1902010 - Frog
   * 1902011 - Turtle
   * 1902012 - Yeti
   */
  def id: Int = {
    if (itemId < 1903000) return itemId - 1901999
    5
  }

  def id_=(newId: Int): Unit = {
    itemId = newId
  }

  def tiredness: Int = _tiredness

  def tiredness_=(newTiredness: Int): Unit = {
    _tiredness = newTiredness
    if (newTiredness < 0) {
      _tiredness = 0
    }
  }

  def incrementAndGetTiredness(): Int = {
    _tiredness += 1
    _tiredness
  }
}

File: engine/engine-model/src/main/java/client/MapleMount.java
package client;

public record MapleMount(int itemId, int skillId, int level, int exp, int tiredness, boolean active) {
   public MapleMount(int itemId, int skillId) {
      this(itemId, skillId, 1, 0, 0, true);
   }

   /**
    * 1902000 - Hog
    * 1902001 - Silver Mane
    * 1902002 - Red Draco
    * 1902005 - Mimiana
    * 1902006 - Mimio
    * 1902007 - Shinjou
    * 1902008 - Frog
    * 1902009 - Ostrich
    * 1902010 - Frog
    * 1902011 - Turtle
    * 1902012 - Yeti
    */
   public Integer id() {
      if (itemId < 1903000) {
         return itemId - 1901999;
      }
      return 5;
   }

   public MapleMount updateItemId(int itemId) {
      return new MapleMount(itemId, skillId, level, exp, tiredness, active);
   }

   public MapleMount setActive(boolean active) {
      return new MapleMount(itemId, skillId, level, exp, tiredness, active);
   }

   public MapleMount updateLevel(int newLevel) {
      return new MapleMount(itemId, skillId, newLevel, exp, tiredness, active);
   }

   public MapleMount updateExp(int newExp) {
      return new MapleMount(itemId, skillId, level, newExp, tiredness, active);
   }

   public MapleMount updateTiredness(int newTiredness) {
      if (newTiredness < 0) {
         newTiredness = 0;
      }
      return new MapleMount(itemId, skillId, level, exp, newTiredness, active);
   }

   public MapleMount incrementTiredness() {
      return new MapleMount(itemId, skillId, level, exp, tiredness + 1, active);
   }
}
